state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1267,20 +1312,22 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 					Some(pool) => pool,
 				};
 
-				used_pools.push(boost_tier);
-
 				pool.provide_funds_for_boosting(boost_id, remaining_amount).map_err(Into::into)
 			})?;
 
+			if !boosted_amount.is_zero() {
+				used_pools.insert(boost_tier, boosted_amount);
+			}
+
 			remaining_amount.saturating_reduce(boosted_amount);
 			total_fee_amount.saturating_accrue(fee);
 
 			if remaining_amount == 0u32.into() {
-				return Ok((used_pools, total_fee_amount));
+				return Ok(BoostOutput { contributions: used_pools, total_fee: total_fee_amount });
```

Comment:
Would just rename `contributions` here to `used_pools` no need for separate names (and used_pools is a little clearer what it's about exactly imo). Especially since we just name it back to used_pools literally right after.

```suggestion
				return Ok(BoostOutput { used_pools, total_fee: total_fee_amount });
```